src/Core/Content/Media/Infrastructure/Path/MediaUrlGenerator.php

```diff
@@ -26,7 +26,10 @@ public function generate(array $paths): array
             if (str_starts_with($value->path, 'http')) {
                 $url = $value->path;
             } else {
-                $url = $this->filesystem->publicUrl($value->path);
+                $pathInfo = pathinfo($value->path);
+                $encodedFilename = rawurlencode($pathInfo['basename']);
+                $encodedPath = ($pathInfo['dirname'] ?? '') . '/' . $encodedFilename;
```

Comment:
```suggestion
                $encodedPath = \sprintf('%s/%s', $pathInfo['dirname'] ?? '', $encodedFilename);
```
I personally think that sprintf is always a bit more clean than native concatination